### PR TITLE
Consolidate API routes and add S3 query file storage

### PR DESCRIPTION
**Pull request by tym-xqo**
- move `v2/reports/<query_name>` endpoints to `<query_name>/docs`
- /docs endpoints are backwards-compatible w/ `v1` API routes, so this removes `v2` and all `v2/results/<query_name>` endpoints
- Upgrade `sqla-raw` to 1.4.0 for `s3fs` support
- Add `s3fs` support to query parsing and discovery/documentation methods

### CODE REVIEW

File: nerium/query.py
Comment (grymoire7) [inline nerium/query.py]: The "rt" may fall back to "rb" but it's not actually supported in this library (or indeed the underlying S3 API), so I think this line is kind of misleading.  See a similar comment on a previous pr [here](https://github.com/nerium-data/sqla-raw/pull/17/files#r995248650).

File: nerium/query.py
Comment (grymoire7) [inline nerium/query.py]: While these lines are not covered (yet) by unit tests, I'm guessing this was maybe tested manually with an S3 query file?

File: nerium/query.py
Comment (tym-xqo) [reply nerium/query.py]: The documentation would seem to be out of date on this point

```
>>> f = s3fs.S3FileSystem()
>>> type(f.open('s3://nerium-test/test.sql', 'rb').read())
<class 'bytes'>
>>> type(f.open('s3://nerium-test/test.sql', 'rt').read())
<class 'str'>
```

But then again, I'm not crazy about relying on an undocumented feature either. I used `rt` here because I was getting TypeErrors from treating the read result as a string later, but so an explicit `str()` call on a binary read is probably better. I'll push that change before this gets merged.

File: nerium/query.py
Comment (tym-xqo) [reply nerium/query.py]: A bunch, yeah. I didn't add unit tests for the S3 stuff because I am a bear of very little brain, and mocks confuse me -- but I hope to noodle it out and get this covered soon. But I've run it with queries on S3 a bunch at this point

File: nerium/query.py
Comment (tym-xqo) [reply nerium/query.py]: Perhaps this is a case of `s3fs` maintainers not understanding the `fsspec` library they're building on top of? It's true that S3 access is always binary, but `fsspec.OpenFile` [allows layering text-mode over it](https://filesystem-spec.readthedocs.io/en/latest/features.html#openfile-instances), using `PickleableTextIOWrapper`. None of this changes that an explicit conversion to text would be better here -- I just wanted to understand how this was working at all, so I dug around a bit

Comment (grymoire7) [reply nerium/query.py]: Ah, thanks!  That makes more sense.  I had looked at the docs and the source for both `s3fs` and `S3File` but I skipped the `fsspec` docs and missed the bit about "Transparent text-mode".  It just looked a bit odd since I knew S3 access to be binary.

**Review by grymoire7 — APPROVED** [review]
👍🏼

File: nerium/query.py
Comment (tym-xqo) [reply nerium/query.py]: Thank YOU! This was all pretty interesting.

Even though it works as originally written, I have gone ahead and changed this to an expressly binary read that I explicitly convert to text, for the sake of clarity for now. I'm also considering opening an issue on `s3fs` to clarify their documentation.

**Pull request merged**
